toolkit/tools/imagegen/configuration/disk.go

```diff
@@ -21,13 +23,73 @@ type Disk struct {
 	RawBinaries        []RawBinary        `json:"RawBinaries"`
 }
 
+// checkOverlappingPartitions checks that start and end positions of the defined partitions don't overlap.
+func checkOverlappingePartitions(disk *Disk) (err error) {
+	partIntervals := [][]uint64{}
+	//convert  partition entries to array of [start,end] locations
+	for _, part := range disk.Partitions {
+		partIntervals = append(partIntervals, []uint64{part.Start, part.End})
+	}
+	//sorting paritions by start position
+	sort.Slice(partIntervals, func(i, j int) bool {
+		return partIntervals[i][0] < partIntervals[j][0]
+	})
+	//confirm each partition ends before the next starts
+	for i := 0; i < len(partIntervals)-1; i++ {
```

Comment:
this can be `for i,_ := range partIntervals {`. Much more golang style.

Reply:
oh, I see you are intending to skip the last entry, could be `range partIntervals[:len(partIntervals) -1 ]` which is a bit of a mess, but still more in the golang style